### ISSUE TITLE
CLI cohort info

### ISSUE THREAD

**Issue by grarco** [anoma/namada-trusted-setup#64]
At the moment the CLI doesn't print a meaningful message when the `join_queue` method fails because of an invalid token for the current cohort. We should provide some info to the user about:

- Cohort he's been assigned to
- Start datetime of the above cohort
- Time left to the above datetime

We can implement this in two ways:

1. Add an extra endpoint to be queried at the beginning of the contribution to retrieve these info
2. Embedded these info directly inside the token

The second option seems to be the best one since it would avoid extra calls to the coordinator server and would allow to implement this feature entirely client-side. To support this, we can create a json like the following:

```json
{
  "start_datetime": "<UtcDateTime>",
  "cohort_duration": "<u64>",
  "cohort_number": "<u64>",
  "token": "<String>"
}
```

This json can then be serialized and encoded `base64` to provide a compact representation tor the CLI input. The CLI can then reconstruct the original data and extract the relevant information for the user.

To better support this the token may become a CLI parameter instead of being asked to the user during the contribution process

**Comment by grarco** [comment]
Actually, since we'll blacklist tokens as per #65, it might be more useful to have an endpoint dedicated to this. By doing so, we should be able to determine in advance both the correct cohort and also whether the token has already been used

**Comment by grarco** [comment]
Actually, we'll skip the endpoint option and stick to the JSON (we won't provide info about the token being blacklisted). In the coordinator server we'll use the `base64` encoded json as the token

**Comment by grarco** [comment]
Closed by #71 and #72
